pkg/discovery/peerlistmgr/peer_list_mgr.go

```diff
@@ -173,12 +173,21 @@ func (m *PeerListManager) ensureConnections() {
 		ctx, cancel := context.WithTimeout(context.Background(), m.connCheckTimeout)
 		conn, err := peer.GetConnection(ctx)
 		cancel()
-		if err == nil && conn.IsActive() {
-			m.logger.Info("Connected to peer", zap.String("host:port", conn.RemotePeerInfo().HostPort))
-			numConnected++
-			if numConnected >= minPeers {
-				return
-			}
+		if err != nil {
+			m.logger.Error("Unable to connect", zap.String("host:port", peer.HostPort()), zap.Duration("connCheckTimeout", m.connCheckTimeout), zap.Error(err))
+			continue
+		}
+
+		if conn.IsActive() == false {
+			m.logger.Info("Connection is not active")
```

Comment:
Is there a way to write a unit test for this condition? Currently it fails code coverage:

https://coveralls.io/builds/13009121/source?filename=pkg%2Fdiscovery%2Fpeerlistmgr%2Fpeer_list_mgr.go

Reply:
conn.IsActive() == false isn't currently exercised in current tests, which caused the decrease in code coverage.

I will push a new commit that structures the code how it was before, which avoids the decrease in code coverage.

This will leave the case where the connection is inactive without any logging (as it currently is).